app/src/main/java/com/example/wrk/CreateAdapter.java

```diff
@@ -115,7 +120,12 @@ public void onClick(View v) {
 
     @RequiresApi(api = Build.VERSION_CODES.N)
     private void queryComponents(WorkoutTemplate template) throws ParseException, JSONException {
-        workoutComponents = new ArrayList<>();     // initialize empty list each time a query is called
+        if (workoutComponents != null) {
+            workoutComponents.clear();
+        }
+        else {
+            workoutComponents = new ArrayList<>();     // initialize empty list each time a query is called
+        }
```

Comment:
sorry once again I don't remember Java, why can't you use clear() always? are there side effects ?

Reply:
If you call clear on a list that hasn't been initialized yet, then it throws a NullPointer error